### DISCUSSION TITLE
2011-12-06 Core team update

### DISCUSSION THREAD

**Discussion by mm-near** [near/community#150] [Core]
# Highlight
New validator assignment change is in testnet (successfully - but with some issues due to database update)

# TL;DR
@pmnoxx - improving network code & documentation, network performance / network rate-limiting / load-testing
@Longarithm - improving fees / debugging performance inconsistencies
@mzhangmzz -   improve chunk fetching / improving block validation error handling
@yordanmadzhunkov - fuzz-testing 
@EgorKulikov - working on different validator assignment 
@abacabadabacaba - better storage analysis (replace Merkle tree with AVL)

# Done:
@pmnoxx - more PRs with documentation improvements and core refactorings to improve code layout and understanding
@pmnoxx - fixing issues discovered by the load tester.

@Longarithm - perf improvements after VecAssembler fix - 2x improvement, but it seems that we didn't charge for it anyways.
@Longarithm - presentation for all hands
@Longarithm - looking at the flaky times of some receipt executions - seems that rocksDB is responsible for them.

@mzhangmzz - Documented ShardLayout and EncodedChunkCache. Will document more files regarding ShardsManager this week as I touch them.
@mzhangmzz  - Tried to reproduce the OOM error in testnet canary before the testnet release, but couldn't. Reverted the change and will test more in mocknet test with Nilolay
@mzhangmzz  - Found a few places for optimization in processing/requesting partial encoded chunks. For example, https://github.com/near/nearcore/pull/5636. Will work on the rest this week
@mzhangmzz  - Thought about a design for a debug page, which will display useful information regarding a node's state for debugging, for example, a chain graph, network info, etc.

@yordanmadzhunkov - fuzzing working with empty block generation - next step is to add more tests and compute testing coverage (both for fuzzing and for regular tests)
@EgorKulikov  - debugging issues related with the new validation algorithm's database update ("Received invalid data"). We should have a better solution for the future upgrades.
@EgorKulikov - looking at the garbage collection issues

@mm-near - Documentation fixit week - added documentation to multiple crates.

**Comment by maxzaver** [comment]
> @Longarithm - perf improvements after VecAssembler fix - 2x improvement, but it seems that we didn't charge for it anyways.

Does it improve module compilation or module loading? @Longarithm CC @mm-near

**Reply by Longarithm** [reply]
> Does it improve module compilation or module loading? 

It improves compilation, [link](https://near.zulipchat.com/#narrow/stream/296237-nearinc.2Fcontract-runtime.2Fprivate/topic/contract.20compilation.20fee/near/263160215)
